api4/channel.go

```diff
@@ -899,6 +900,79 @@ func getChannelsForTeamForUser(c *Context, w http.ResponseWriter, r *http.Reques
 	}
 }
 
+func getChannelsForUser(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	if !c.App.SessionHasPermissionToUser(*c.AppContext.Session(), c.Params.UserId) {
+		c.SetPermissionError(model.PermissionEditOtherUsers)
+		return
+	}
+
+	query := r.URL.Query()
+	lastDeleteAt, nErr := strconv.Atoi(query.Get("last_delete_at"))
+	if nErr != nil {
+		lastDeleteAt = 0
+	}
+	if lastDeleteAt < 0 {
+		c.SetInvalidURLParam("last_delete_at")
+		return
+	}
+
+	pageSize := 100
+	fromChannelID := ""
+	// We have to write `[` and `]` separately because we want to stream the response.
+	// The internal API is paginated, but the client always needs to get the full data.
```

Comment:
@agnivade @streamer45 As we need data of channels across team we had two options either load channels when user opens Channel switcher or Load all the channels on app load. So to make UX better [as discussed here](https://community-daily.mattermost.com/core/pl/16otwhqhpfn58ym9h3tjfoiiso) we have taken middle path of loading all the channels user is part of across teams on app load. As generally user is looking to switch channel instead of joining.

Reply:
Okay cool, this only returns channels the user is part of (has joined). Was slightly worried for a second :)